src/communication/network_interface_sr.c

```diff
@@ -7268,6 +7269,208 @@ slogtb_dump_trantable (THREAD_ENTRY * thread_p, unsigned int rid, char *request,
   db_private_free_and_init (thread_p, buffer);
 }
 
+/*
+ * smmon_get_module_info -
+ *
+ * return:
+ *
+ *   rid(in):
+ *   request(in):
+ *   reqlen(in):
+ */
+void
+smmon_get_module_info (THREAD_ENTRY * thread_p, unsigned int rid, char *request, int reqlen)
+{
+  char *buffer, *ptr;
+  int size = 0;
+  OR_ALIGNED_BUF (OR_INT_SIZE + OR_INT_SIZE) a_reply;
+  char *reply = OR_ALIGNED_BUF_START (a_reply);
+  int error = NO_ERROR;
+  MMON_MODULE_INFO info;
+  MMON_COMP_INFO *c_info;
+  MMON_SUBCOMP_INFO *s_info;
+  int module_index = 0;
+
+  ptr = or_unpack_int (request, &module_index);
+
+  error = mmon_aggregate_module_info (info, module_index);
+
+  if (error == NO_ERROR)
+    {
+      /* before send information 
+       * 1) check size
+       * 2) allocate buffer
+       * 3) packing information 
+       * 4) deallocate MMON_MODULE_INFO */
+
+      /* 1) check size 
+       *    - calculate buffer size of packed values
+       *    - server info -> module info -> component info w/ subcomponent info */
+      size += or_packed_string_length (info.server_info.name, NULL);
+      size += OR_INT64_SIZE;
+
+      size += or_packed_string_length (info.name, NULL);
+      size += OR_INT64_SIZE * 3 + OR_INT_SIZE * 2;
```

Comment:
I think protocol is required to be described for me to review these size

And `OR_INT64_SIZE * 3 + OR_INT_SIZE * 2` can be defined as const variable.